pkg/nodes/enricher/enricher_impl.go

```diff
@@ -49,7 +49,14 @@ func (e *enricherImpl) RemoveNodeIntegration(id string) {
 
 // EnrichNode enriches a node with the integration set present.
 func (e *enricherImpl) EnrichNode(node *storage.Node) error {
+	// Clear any pre-existing notes, as it will all be filled here.
+	// Note: this is valid even if node.Notes is nil.
```

Comment:
> as it will all be filled here.

Sanity check: Is it possible we will add notes to nodes outside the Enrich() pipeline? Good to go either way. I am checking because I want to understand better why you decided to reset all notes. Maybe renaming the field to `EnrichingNotes` would make things more straightforward, but I digress.

Reply:
Thinking about it, I don't think so. I know we currently don't, but of course that is not future proof. However, `EnrichNode` is meant to be the sole enricher of the node and the notes are part of the node, so I think it makes sense it is the only source of notes

Reply:
Sounds good. Thanks.